src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterStepExtensions.cs

```diff
@@ -14,16 +14,19 @@ public static class PackageUpdaterStepExtensions
 
         public static void AddPackageUpdaterStep(this IExtensionServiceCollection services)
         {
+
             if (services is null)
             {
                 throw new System.ArgumentNullException(nameof(services));
             }
 
             services.Services.AddUpgradeStep<PackageUpdaterPreTFMStep>();
             services.Services.AddUpgradeStep<PackageUpdaterStep>();
+            services.Services.AddSingleton<IDependencyAnalyzerRunner, DependencyAnalyzerRunner>();
```

Comment:
Nit-pick: Transient is probably fine for this since it doesn't need to preserve state and isn't particularly expensive to new up.

Reply:
Changed